File: backend/models/saved.py
"""Saved-business domain models for the Vantage API.

Defines lightweight models for tracking which businesses a user has
saved/bookmarked, and the result of a save/unsave toggle operation.
"""
from datetime import datetime
from pydantic import BaseModel

class SavedRecord(BaseModel):
    """A record linking a user to a saved business.

    Attributes:
        user_id (str): ID of the user who saved the business.
        business_id (str): ID of the saved business.
        created_at (datetime): Timestamp when the save occurred.
    """
    user_id: str
    business_id: str
    created_at: datetime

class SavedMutationResult(BaseModel):
    """Result of a save or unsave operation.

    Attributes:
        business_id (str): ID of the affected business.
        saved (bool): True after saving, False after removing.
    """
    business_id: str
    saved: bool

File: backend/src/models/saved.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SavedRecord {
    #[serde(rename = "_id", skip_serializing_if = "Option::is_none")]
    pub id: Option<String>,
    pub user_id: String,
    pub business_id: String,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Serialize)]
pub struct SavedMutationResult {
    pub saved: bool,
    pub business_id: String,
}
